src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/UnsafeRelaxedJavaScriptEncoder.cs

```diff
@@ -140,21 +162,31 @@ public override unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf
                 int idx = 0;
 
 #if NETCOREAPP
-                if (Sse2.IsSupported)
+                if (Sse2.IsSupported || AdvSimd.Arm64.IsSupported)
                 {
                     sbyte* startingAddress = (sbyte*)ptr;
                     while (utf8Text.Length - 16 >= idx)
                     {
                         Debug.Assert(startingAddress >= ptr && startingAddress <= (ptr + utf8Text.Length - 16));
 
                         // Load the next 16 bytes.
-                        Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                        Vector128<sbyte> sourceValue;
+                        bool containsNonAsciiBytes;
 
                         // Check for ASCII text. Any byte that's not in the ASCII range will already be negative when
                         // casted to signed byte.
-                        int index = Sse2.MoveMask(sourceValue);
+                        if (Sse2.IsSupported)
+                        {
+                            sourceValue = Sse2.LoadVector128(startingAddress);
+                            containsNonAsciiBytes = Sse2.MoveMask(sourceValue) != 0;
+                        }
+                        else
+                        {
+                            sourceValue = AdvSimd.LoadVector128(startingAddress);
+                            containsNonAsciiBytes = AdvSimd.Arm64.MinAcross(sourceValue).ToScalar() < 0;
```

Comment:
>MinAcross [](start = 66, length = 9)

I would recommend to create a method in `AdvSimdHelpers` and name it something like `ContainsAsciiBytes()` and mark it `[AggressiveInlining]`. Not only we will have this implementation at one place, but in future if we find out that there is an optimal way to do this operation, we will just need to change in one place.